2.Project2/datasource.cpp

```diff
@@ -91,8 +97,32 @@ void DataSource::dataReadFinished()
 
             addJoke(joke);
         }
-
+        mRequestsToMake = 0;
+#endif
         //Clear the buffer
         mDataBuffer->clear();
+        if(mRequestsToMake > 0){
+            --mRequestsToMake;
+            mRequestState = REQUESTS_ONGOING;
+            emit requestCompleted();
```

Comment:
Your logic seems to rely on the requestCompleted signal to fetch jokes one after the other. Please also implement the logic to just fetch jokes in a simple loop. Keep your logic in place, and provide the ability to to switch between the two through some macro definition.

Reply:
I have added supported for a looped approach. However, this approach can be blocking and might not yield good results in the case of a high latency network during a request.